app/models/status.rb

```diff
@@ -141,7 +141,7 @@ def searchable_by(preloaded = nil)
       ids += preloaded.reblogs[id] || []
     end
 
-    ids.uniq
+    Account.where('id in (?)', ids.uniq).filter(&:local?).map(&:id)
```

Comment:
This is an n+1 query which is why the indexing becomes slower. It also loads all account objects into memory before filtering them with Ruby instead of only selecting local accounts with SQL.

Instead, you can change the `crutch` definitions in `chewy/statuses_index.rb` to filter for local accounts in the first place.

For example:

```ruby
data = ::Mention.where(status_id: collection.map(&:id)).where(account: Account.local).pluck(:status_id, :account_id)
```